### ISSUE TITLE
Connect from EC2 using Ec2InstanceMetadata credential_source

### ISSUE THREAD

**Issue by benjamin-fenker** [aws/aws-toolkit-vscode#1007]
-   OS: Linux
-   VS Code version: 1.44.0-insider
-   AWS Toolkit extension version:1.8.0

**Question**

I'm running VS code on an EC2 instance with an attached role and looking to configure the plugin to use this role for AWS access. My config file looks like:
```
[profile user]
role_arn=arn:aws:iam::XXXXXXXX:role/XXXXXXXX
credential_source=Ec2InstanceMetadata
```
but getting an error that `source_profile` is not set using profile user. Documentation suggests that either `credential_source` or `source_profile` are required. Is this a limitation of the extension or is something configured incorrectly on my end?

**Comment by awschristou** [comment]
Hi @benjamin-fenker , the toolkit currently doesn't have as much EC2 support as it could. Could you try using a `credential_process` approach instead of using `credential_source` as a workaround? This is discussed in https://github.com/aws/aws-toolkit-vscode/issues/695#issuecomment-516163498

**Comment by justinmk3** [comment]
Related : 

- https://github.com/aws/aws-sdk-js/issues/2071
- https://github.com/aws/aws-sdk-js/issues/1916

**Comment by cliv** [comment]
Thanks to @awschristou i was able to get this working.

https://gist.github.com/cliv/59eafdc0654c8ce4d9803ae1bb465884 contains the recipe i used to allow the use of an EC2 instance IAM role in the AWS Toolkit while connected to a running EC2 instance via VSCode's ms-vscode-remote.remote-ssh extension.

**Comment by justinmk3** [comment]
We merged https://github.com/aws/aws-toolkit-vscode/pull/1833 which adds support for [credential_source](https://docs.aws.amazon.com/sdkref/latest/guide/setting-global-credential_source.html). If you want to try the prerelease we'd love your feedback: 

1. Download and extract [awstoolkit-0ea03d.zip](https://github.com/aws/aws-toolkit-vscode/files/6737111/awstoolkit-0ea03d.zip)
2. From the VSCode command list (`ctrl-shift-p` or `cmd-shift-p`) run `Extensions: Install from VSIX...` to install the `*.vsix` file.
3. Reload VSCode and verify that the Toolkit has the new version (not the older version).

**Comment by justinmk3** [comment]
https://github.com/aws/aws-toolkit-vscode/pull/1833 adds support for [credential_source](https://docs.aws.amazon.com/sdkref/latest/guide/setting-global-credential_source.html), released in Toolkit 1.26.
